### ISSUE TITLE
bug: Dynamic import / lazyloading randomly not working in production

### ISSUE THREAD

**Issue by TrimA74** [ionic-team/stencil#3281]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.14.2

### Current Behavior

I have web application using StencilJS in production and for some reason sometimes we got Bugsnag errors when a client try to dynamically import a module, i checked the module is existing in production and available 

`TypeError: Failed to fetch dynamically imported module: https://DOMAIN.com/shop/build/p-62cc5219.entry.js`

### Expected Behavior

This should not happen, dynamic import should work properly everytime ?

### Steps to Reproduce

I never succeed to reproduce the same problem but i still receive bugsnag error from production application and it has impact  on my clients 

### Code Reproduction URL

https://github.com/ionic-team/stencil

### Additional Information

https://github.com/ionic-team/stencil/blob/main/src/client/client-load-module.ts

Found the source file where the dynamic import is made

**Comment by rwaskiewicz** [comment]
Hey @TrimA74 👋 

Thanks for the report. Unfortunately, there isn't enough information here for us to look into this issue further.  I don't doubt you that you're running into file availability issues, however we don't have a good way to determine if this is a Stencil issue, an issue with the hosting service you're using or something else based on the information provided. 

If you're able to look at logs for your hosting service, that might help narrow this down further.  I'm going mark this issue as 'awaiting reply' to see if there's anything you can do to validate this is a Stencil issue.  If I don't hear from you, I'll close this issue in about a week's time.  Thanks again!

**Comment by TrimA74** [comment]
Hi @rwaskiewicz , thanks for the response ! 

So for example i took a recent example, error juste happened 18 minutes ago on this site : https://etickets.infomaniak.com/shop/3tmijx8AXw/

The error is : TypeError: Failed to fetch dynamically imported module: https://etickets.infomaniak.com/shop/build/p-e02612e8.entry.js

And as you can see there is no problem on the hosting service side to fetch the file : https://etickets.infomaniak.com/shop/build/p-e02612e8.entry.js

Other example : 

https://etickets.infomaniak.com/shop/28W6Fj03K9/

TypeError: Failed to fetch dynamically imported module: https://etickets.infomaniak.com/shop/build/p-b0579a8a.entry.js

**Comment by rwaskiewicz** [comment]
Hey @TrimA74,

For the first case there, I do see some spikes in not being able to load certain files, which aligns with what you're saying about not being able to fetch the module. Unfortunately there's still no great way to determine if this is a transient error with your hosting provider or Stencil itself. If you could find a way to reproduce this locally that'd give us a better chance at root causing any Stencil-related issues.   

For that second case (labeled 'other example'), I actually don't see a `p-b0579a8a.entry.js` file being downloaded in the developer tooling, can you verify that file's a part of the current production build? I also cannot access that file, but if the hash has changed as a result of a new build going to prod, that would explain why
![Screen Shot 2022-03-15 at 7 49 30 AM](https://user-images.githubusercontent.com/1930213/158371813-579a0795-5269-4749-89aa-c461aba124dc.png)

**Comment by rwaskiewicz** [comment]
> What do you mean when you say "I do see some spikes in not being able to load certain files" ?

I went to https://etickets.infomaniak.com/shop/3tmijx8AXw/ and loaded the page several times without any cache enabled. I was able to observe some slowness in resolving `[p-e02612e8.entry.js](https://etickets.infomaniak.com/shop/build/p-e02612e8.entry.js)`, which I was saying aligns with what you are saying about the file not being found.  

Unfortunately that doesn't tell us for sure if its the infrastructure that's serving the file that's failing or Stencil itself.  If you were able to create a minimal reproduction case with the production build, that'd help us greatly in narrowing the issue down.
